aiosqlite/cursor.py

```diff
@@ -12,18 +12,30 @@ class Cursor:
     def __init__(self, conn: "Connection", cursor: sqlite3.Cursor) -> None:
         self._conn = conn
         self._cursor = cursor
+        self.__chunk: Optional[Iterator[sqlite3.Row]]
+        self.iter_chunk_size = 1024
```

Comment:
This seems like it's potentially a lot to buffer in memory for each cursor. Did you do performance tuning/testing to arrive at this value, or was it arbitrarily chosen? I think it would also be useful to have this as an optional property set on the Connection class, so that it could be set once and reused for each cursor on that connection. I would also probably default to a smaller value, like 64, since it's possible that each row could be quite large.

Reply:
Not really, it was just a guess, but I agree with your concerns, and in fact 64 turns out to be a nice balance between RAM / performance.

On my modest hardware I got these perf results (the original did 36k iters / 2s):

Chunk size | Iterations | Duration | Rate | Gain | Gain over the original
-- | -- | -- | -- | -- | --
16 | 410,705 | 2.0s | 205,317 |   | ×11
32 | 644,097 | 2.0s | 321,992 | 57% | ×18
64 | 893,560 | 2.0s | 446,702 | 39% | ×24
128 | 1,111,297 | 2.0s | 555,551 | 24% | ×30
256 | 1,275,905 | 2.0s | 637,841 | 15% | ×35
512 | 1,380,353 | 2.0s | 690,056 | 8% | ×38
1024 | 1,447,937 | 2.0s | 723,842 | 5% | ×39